tests/backends/argo/test_component.py

```diff
@@ -0,0 +1,115 @@
+import re
+from contextlib import nullcontext as does_not_raise
+
+import pytest
+from argo_workflows.model.io_argoproj_workflow_v1alpha1_artifact import (
+    IoArgoprojWorkflowV1alpha1Artifact,
+)
+from argo_workflows.model.io_argoproj_workflow_v1alpha1_inputs import (
+    IoArgoprojWorkflowV1alpha1Inputs,
+)
+from argo_workflows.model.io_argoproj_workflow_v1alpha1_template import (
+    IoArgoprojWorkflowV1alpha1Template,
+)
+
+from exedra.backends.argo import component
+
+
+@pytest.fixture
+def mock_component(request):
+    return component.Component(
+        template=IoArgoprojWorkflowV1alpha1Template(
+            name="test_component",
+            inputs=IoArgoprojWorkflowV1alpha1Inputs(
+                artifacts=[
+                    IoArgoprojWorkflowV1alpha1Artifact(name=key)
+                    for key in request.param
+                ]
+            ),
+        )
+    )
+
+
+@pytest.fixture
+def expectation(request):
+    return (
+        pytest.raises(TypeError, match=re.escape(request.param))
+        if request.param is not None
+        else does_not_raise()
+    )
+
+
+@pytest.mark.parametrize(
+    "mock_component,expectation",
+    [(["input1"], "test_component() got an unexpected keyword argument 'input2'")],
+    indirect=True,
+)
+def test_invalid_arguments(mock_component, expectation):
+    with expectation:
+        mock_component(input2=2)
+
+
+@pytest.mark.parametrize(
+    "mock_component,args,kwargs,expectation",
+    [
+        ([], [], {}, None),
```

Comment:
Is this necessary? If anything, I don't think it should fall under `test_too_many_arguments`. It's the happy path, and should eventually do more attribute checks on this.